client/my-sites/checkout/composite-checkout/components/prepurchase-notices/akismet-product-overlaps-owned-product-notice.tsx

```diff
@@ -0,0 +1,55 @@
+import { useTranslate } from 'i18n-calypso';
+import { FunctionComponent } from 'react';
+import { useSelector } from 'react-redux';
+import { getProductBySlug } from 'calypso/state/products-list/selectors';
+import { getUserPurchases } from 'calypso/state/purchases/selectors';
+import PrePurchaseNotice from './prepurchase-notice';
+
+import './style.scss';
+
+type Props = {
+	overlapingProductSlug: string;
+	cartProductSlug: string;
+};
+
+const AkismetProductOverlapsOwnedProductNotice: FunctionComponent< Props > = ( {
+	overlapingProductSlug,
+	cartProductSlug,
+} ) => {
+	const translate = useTranslate();
+	const products = useSelector( ( state ) => ( {
+		overlaping: getProductBySlug( state, overlapingProductSlug ),
+		cart: getProductBySlug( state, cartProductSlug ),
+	} ) );
+	const purchases = useSelector( getUserPurchases );
+	const purchase = Array.isArray( purchases )
+		? purchases.find( ( p ) => p.productSlug === products.overlaping?.product_slug )
+		: null;
+	const purchaseId = purchase?.id;
+	const subscriptionUrl = purchaseId
+		? `/me/purchases/siteless.akismet.com/${ purchaseId }`
+		: '/me/purchases/';
+
+	const message = translate(
+		'You current own {{oldProduct/}}. The plan you are about to purchase {{newProduct/}} contains all the same functionality and benefits of your current plan. Consider {{link}}removing your {{oldProduct/}} subscription{{/link}} to avoid paying for both plans.',
```

Comment:
I wonder if we should use string interpolation instead of component interpolation for these variables. Using `components` implies that each variable is a React component or HTML tag, but these just look like strings, which means we can use `args` instead (see [the documentation](https://github.com/Automattic/wp-calypso/blob/trunk/packages/i18n-calypso/README.md#string-substitution) for using variables in `translate`).

Also there's a typo here. It should be "You currentLY own..."
```suggestion
		'You currently own %(oldProduct)s. The plan you are about to purchase %(newProduct)s contains all the same functionality and benefits of your current plan. Consider {{link}}removing your %(oldProduct)s subscription{{/link}} to avoid paying for both plans.',
```

Reply:
Thanks, it makes sense to change it from component to arg. I've also reworked the text in general as per @jboland88 suggestions.